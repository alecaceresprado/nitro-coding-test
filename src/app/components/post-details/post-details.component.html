<div class="post-details">
  <div class="post-details__title">Post details</div>
  <div class="post-details__header" *ngIf="!!post; else postZrp">
    Post number {{ post.id }} | {{ postDate }} <br />
    Written in {{ post.location }}, by {{ post.author }}
    <div class="post-details__content">
      <div class="post-details__content-title">content</div>
      <div class="post-details__content-text">{{ post.text }}</div>
    </div>
    <button class="post-details__button" (click)="toggleEditMode()">
      edit
    </button>
  </div>
  <div class="post-details__edit-form">
    <app-edit-form
      *ngIf="editMode"
      (success)="toggleEditMode()"
      [post]="post"
    ></app-edit-form>
  </div>
</div>

<ng-template #postZrp>
  <div class="post-details__header">
    Please select a post from the list on your left
  </div>
</ng-template>
