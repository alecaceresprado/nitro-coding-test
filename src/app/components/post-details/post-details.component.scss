@import "../../../styles/vars";

.post-details {
  &__title {
    @include sectionTitle();
  }
  &__header {
    margin: 1rem;
  }

  &__content {
    @include layout();
    padding: 1rem;
  }
  &__content-title {
    width: fit-content;
    position: relative;
    top: -1.6rem;
    background: white;
    margin-bottom: -1.25rem;
    padding: 0 0.25rem;
  }

  &__button {
    float: right;
    margin-right: 0.5rem;
    border: none;
    width: 3rem;
    cursor: pointer;
    height: 1.5rem;
  }

  &__edit-form {
    position: relative;
    top: 2rem;
    margin: 1rem;
  }
}
