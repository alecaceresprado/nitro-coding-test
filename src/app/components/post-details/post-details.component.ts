import { Component, Input, OnChanges } from '@angular/core';

import { Post } from '@models';

@Component({
  selector: 'app-post-details',
  templateUrl: './post-details.component.html',
  styleUrls: ['./post-details.component.scss'],
})
export class PostDetailsComponent implements OnChanges {
  @Input() public post: Post;

  public editMode = false;

  public ngOnChanges() {
    this.editMode = false;
  }

  public get postDate(): string {
    const postDate = new Date(Number(this.post.time));
    return `${postDate.getFullYear()}/${postDate.getMonth()}/${postDate.getDay()}`;
  }

  public toggleEditMode(): void {
    this.editMode = !this.editMode;
  }
}
