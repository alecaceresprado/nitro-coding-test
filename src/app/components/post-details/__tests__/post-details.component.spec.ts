import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { PostDetailsComponent } from '../post-details.component';

describe('PostDetailsComponent', () => {
  let component: PostDetailsComponent;
  let fixture: ComponentFixture<PostDetailsComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [PostDetailsComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PostDetailsComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render - zrp', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should render - post', () => {
    component.post = {
      id: 1,
      location: 'San Francisco',
      time: '1552657573',
      author: 'Happy User',
      text:
        'Proper PDF conversion ensures that every element of your document remains just as you left it.',
    };
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
