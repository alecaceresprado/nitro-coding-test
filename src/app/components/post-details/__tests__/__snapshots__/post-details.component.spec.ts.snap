// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetailsComponent should render - post 1`] = `
<app-post-details
  editMode="false"
  post={[Function Object]}
>
  <div
    class="post-details"
  >
    <div
      class="post-details__title"
    >
      Post details
    </div>
    
    <div
      class="post-details__header"
    >
       Post number 1 | 1970/0/1 
      <br />
       Written in San Francisco, by Happy User 
      <div
        class="post-details__content"
      >
        <div
          class="post-details__content-title"
        >
          content
        </div>
        <div
          class="post-details__content-text"
        >
          Proper PDF conversion ensures that every element of your document remains just as you left it.
        </div>
      </div>
      <button
        class="post-details__button"
      >
         edit 
      </button>
    </div>
    <div
      class="post-details__edit-form"
    >
      
    </div>
  </div>
</app-post-details>
`;

exports[`PostDetailsComponent should render - zrp 1`] = `
<app-post-details
  editMode="false"
>
  <div
    class="post-details"
  >
    <div
      class="post-details__title"
    >
      Post details
    </div>
    
    <div
      class="post-details__header"
    >
       Please select a post from the list on your left 
    </div>
    <div
      class="post-details__edit-form"
    >
      
    </div>
  </div>
</app-post-details>
`;
